projects/packages/my-jetpack/src/products/class-hybrid-product.php

```diff
@@ -67,11 +67,10 @@ public static function activate_plugin() {
 	/**
 	 * Activates the product. If the Hybrid product has declared a jetpack module name, let's try to activate it if Jetpack plugin is active
 	 *
-	 * @return boolean|\WP_Error
+	 * @param bool|WP_Error $product_activation Is the result of the top level activation actions. You probably won't do anything if it is an WP_Error.
+	 * @return bool|WP_Error
 	 */
-	public static function activate() {
-
-		$product_activation = parent::activate();
+	public static function do_product_specific_activation( $product_activation ) {
 
 		if ( is_wp_error( $product_activation ) ) {
 			return $product_activation;
```

Comment:
Not related to the PR but I suspect the `Jetpack` class reference would need a namespace on L81.

Reply:
We possibly need to use `Modules` class on L81 as well.

Reply:
We possibly need to use `Modules` class on L81 as well.

Reply:
> Not related to the PR but I suspect the Jetpack class reference would need a namespace on L81.

Yes, I noticed that while working on #24335 and fixed it there (these 2 PRs will conflict as soon as one of them gets merged, but that's ok, I'm expecting it)